File: examples/selenium-java-01-search/input.java
package com.acme.shop.tests;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ProductSearchTest {

    private WebDriver driver;
    private WebDriverWait wait;

    @BeforeEach
    void setUp() {
        driver = new ChromeDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://shop.acme.test");
    }

    @AfterEach
    void tearDown() {
        driver.quit();
    }

    @Test
    void searchReturnsMatchingProducts() throws InterruptedException {
        WebElement searchBox = driver.findElement(By.id("site-search"));
        searchBox.sendKeys("linen");
        driver.findElement(By.xpath("//header/div[2]/form/button")).click();

        Thread.sleep(2000);

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".results-grid")));

        List<WebElement> results = driver.findElements(By.cssSelector(".results-grid .product-card"));
        assertTrue(results.size() >= 1, "Expected at least one result");

        WebElement firstTitle = driver.findElements(By.cssSelector(".product-card h3")).get(0);
        assertTrue(firstTitle.getText().toLowerCase().contains("linen"));
    }

    @Test
    void emptySearchShowsHint() throws InterruptedException {
        driver.findElement(By.xpath("//header/div[2]/form/button")).click();
        Thread.sleep(1000);

        WebElement hint = driver.findElement(By.xpath("//div[contains(@class,'search-hint')]/span[2]"));
        assertEquals("Please enter a search term", hint.getText());
    }
}

File: examples/selenium-python-01-login/input.py
import time

import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait


@pytest.fixture
def driver():
    drv = webdriver.Chrome()
    drv.implicitly_wait(5)
    yield drv
    drv.quit()


@pytest.fixture
def logged_in_driver(driver):
    driver.get("https://hr.beacon.test/login")
    driver.find_element(By.ID, "email").send_keys("hr-admin@beacon.test")
    driver.find_element(By.ID, "password").send_keys("Sup3rSecret!")
    driver.find_element(By.XPATH, "//form//button[@type='submit']").click()
    time.sleep(2)
    return driver


def test_user_can_log_in(driver):
    driver.get("https://hr.beacon.test/login")
    driver.find_element(By.ID, "email").send_keys("hr-admin@beacon.test")
    driver.find_element(By.ID, "password").send_keys("Sup3rSecret!")
    driver.find_element(By.XPATH, "//form//button[@type='submit']").click()

    WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.CSS_SELECTOR, ".dashboard-greeting"))
    )

    greeting = driver.find_element(By.CSS_SELECTOR, ".dashboard-greeting").text
    assert greeting == "Welcome back, HR Admin"


def test_dashboard_shows_team_count(logged_in_driver):
    cards = logged_in_driver.find_elements(By.CSS_SELECTOR, ".kpi-card")
    team_card = cards[0]
    count = team_card.find_element(By.CSS_SELECTOR, ".kpi-value").text
    assert int(count) >= 1
